GCRCatalogs/dc2_metacal.py

```diff
@@ -0,0 +1,102 @@
+    use_cache        (bool): Cache read data in memory
+    is_dpdd          (bool): File are already in DPDD-format.  No translation.
+
+    Attributes
+    ----------
+    base_dir          (str): The directory of data files being served
+
+    Notes
+    -----
+    """
+    # pylint: disable=too-many-instance-attributes
+
+    FILE_DIR = os.path.dirname(os.path.abspath(__file__))
+    FILE_PATTERN = r'metacal_tract_\d+\.parquet$'
+    SCHEMA_FILENAME = 'schema.yaml'
+    META_PATH = os.path.join(FILE_DIR, 'catalog_configs/_dc2_metacal_meta.yaml')
+
+
+    def _subclass_init(self, **kwargs):
+        """
+        Overrides default init method to apply various corrections to the catalog
+        """
+        # Default values of reader parameters
+        self._bands = kwargs.get('bands', 'riz')
+
+        self._flux_scaling = 1.0
+        if kwargs.get('fix_metacal_test3'):
+            # In Run 1.2i metacal_test3, the fluxes returned by metacal are not
+            # properly scaled by the pixel size, it's necessary to apply a
+            # 0.2**2 correction factor
+            self._flux_scaling /= 0.2**2
+
+        super(DC2MetacalCatalog, self)._subclass_init(**kwargs)
+
+
+    def _generate_modifiers(self, dm_schema_version=3):
+        """Creates a dictionary relating native and homogenized column names
+
+        Args:
+            dm_schema_version (int): DM schema version (1, 2, or 3)
+
+        Returns:
+            A dictionary of the form {<homogenized name>: <native name>, ...}
+        """
+
+        modifiers = {
+            'objectId': 'id',
+            'mcal_psf_g1': 'mcal_psf_g1_mean',
+            'mcal_psf_g2': 'mcal_psf_g2_mean',
+            'mcal_T_psf' : 'mcal_psf_T_mean',
+            'mcal_flags' : 'mcal_flags'
+        }
+
+        # Additional metacal values and their variants
+        for variant in ['', '_1p', '_1m', '_2p', '_2m']:
+            # Shape
+            modifiers['mcal_g1{}'.format(variant)] = 'mcal_gauss_g1{}'.format(variant)
+            modifiers['mcal_g2{}'.format(variant)] = 'mcal_gauss_g2{}'.format(variant)
+            # Size
+            modifiers['mcal_T{}'.format(variant)] = 'mcal_gauss_T{}'.format(variant)
+            # SNR
+            modifiers['mcal_s2n{}'.format(variant)] = 'mcal_gauss_s2n{}'.format(variant)
+
+            # Adds band dependent info and their variants
+            for band in self._bands:
+                modifiers['mcal_flux_{}{}'.format(band, variant)] = (
+                    lambda x: x * self._flux_scaling,
+                    'mcal_gauss_flux_{}{}'.format(band, variant)
+                )
+                modifiers['mcal_flux_err_{}{}'.format(band, variant)] = (
+                    lambda x: x * self._flux_scaling,
+                    'mcal_gauss_flux_err_{}{}'.format(band, variant)
+                )
+                modifiers['mcal_mag_{}{}'.format(band, variant)] = (
+                    lambda x: -2.5 * np.log10(x * self._flux_scaling) + 27.0,
```

Comment:
Define `27.0` in a named variable somewhere to document where it comes from (and if we ever might have to change it).

For the standard DM catalogs I'm directly reading the calibration information.  Does metacal explicitly use `27.0` as a standard reference zeropoint?

Reply:
Yes, according to @esheldon , I'm  going to define a METACAL_ZEROPOINT variable and set it to 27.0